From 07ec96afaa34d95384df1560347330e8d9095c60 Mon Sep 17 00:00:00 2001
From: Hari Kanigeri <h-kanigeri2@ti.com>
Date: Fri, 31 Jul 2009 22:52:23 -0500
Subject: [PATCH] SYSLINK:ipc-sys manager structure fixes

This patch fixes the sys manager structure misalignment
and a compile warning in nameserver.c

Signed-off-by: Suman Anna <suman-anna@ti.com>
---
 arch/arm/plat-omap/include/syslink/_sysmgr.h   |   12 ++++++++++++
 drivers/dsp/syslink/multicore_ipc/nameserver.c |    1 -
 drivers/dsp/syslink/multicore_ipc/platform.c   |    4 +---
 3 files changed, 13 insertions(+), 4 deletions(-)

diff --git a/arch/arm/plat-omap/include/syslink/_sysmgr.h b/arch/arm/plat-omap/include/syslink/_sysmgr.h
index f7faa22..58fbdd3 100644
--- a/arch/arm/plat-omap/include/syslink/_sysmgr.h
+++ b/arch/arm/plat-omap/include/syslink/_sysmgr.h
@@ -18,6 +18,18 @@
 #ifndef __SYSMGR_H_
 #define __SYSMGR_H_
 
+/* Structure to retrieve the scalability proc info from the slave */
+struct sysmgr_proc_config {
+	u32 proc_id;
+	u32 use_notify;
+	u32 use_messageq;
+	u32 use_heapbuf;
+	u32 use_frameq;
+	u32 use_ringio;
+	u32 use_listmp;
+	u32 use_nameserver;
+	u32 boot_mode;
+};
 
 /* Function to set the boot load page address for a slave */
 void sysmgr_set_boot_load_page(u16 proc_id, u32 boot_load_page);
diff --git a/drivers/dsp/syslink/multicore_ipc/nameserver.c b/drivers/dsp/syslink/multicore_ipc/nameserver.c
index 9199339..1a57ee4 100644
--- a/drivers/dsp/syslink/multicore_ipc/nameserver.c
+++ b/drivers/dsp/syslink/multicore_ipc/nameserver.c
@@ -264,7 +264,6 @@ int nameserver_setup(void)
 error:
 	kfree(list);
 	printk(KERN_ERR "nameserver_setup failed, retval: %x\n", retval);
-exit:
 	return retval;
 }
 EXPORT_SYMBOL(nameserver_setup);
diff --git a/drivers/dsp/syslink/multicore_ipc/platform.c b/drivers/dsp/syslink/multicore_ipc/platform.c
index b1576ba..5c198c9 100644
--- a/drivers/dsp/syslink/multicore_ipc/platform.c
+++ b/drivers/dsp/syslink/multicore_ipc/platform.c
@@ -236,7 +236,6 @@ struct platform_messageq_transportshm_params {
 };
 
 struct platform_proc_config_params {
-	u32 reserved1;
 	u32 use_notify;
 	u32 use_messageq;
 	u32 use_heapbuf;
@@ -244,7 +243,6 @@ struct platform_proc_config_params {
 	u32 use_ring_io;
 	u32 use_listmp;
 	u32 use_nameserver;
-	u32 reserved2;
 };
 
 /** ============================================================================
@@ -321,7 +319,7 @@ void *platform_sm_heap_phys_addr;
 /*!
  *  @brief  Scalability info
  */
-struct platform_proc_config_params pc_params;
+struct sysmgr_proc_config pc_params;
 
 /* =============================================================================
  * APIS
-- 
1.5.4.3

